File: frontend/src/types/flight.ts
export type FlightStatus =
  | "scheduled"
  | "active"
  | "landed"
  | "cancelled"
  | "diverted"
  | "delayed"
  | "unknown";

export type FlightDirection = "departures" | "arrivals";

export interface Airport {
  iata: string;
  name: string | null;
}

export interface Flight {
  flight_number: string;
  airline: string | null;
  origin: Airport;
  destination: Airport;
  scheduled_time: string | null;
  estimated_time: string | null;
  actual_time: string | null;
  status: FlightStatus;
  terminal: string | null;
  gate: string | null;
}

export interface FlightsResponse {
  airport: string;
  type: FlightDirection;
  fetched_at: string;
  cached: boolean;
  flights: Flight[];
}

export interface FlightLookupResponse {
  flight_number: string;
  fetched_at: string;
  cached: boolean;
  flights: Flight[];
}

File: backend/app/models/flight.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel

FlightStatus = Literal[
    "scheduled", "active", "landed", "cancelled", "diverted", "delayed", "unknown"
]

FlightType = Literal["departures", "arrivals"]


class Airport(BaseModel):
    iata: str
    name: str | None = None


class Flight(BaseModel):
    flight_number: str
    airline: str | None = None
    origin: Airport
    destination: Airport
    scheduled_time: datetime | None = None
    estimated_time: datetime | None = None
    actual_time: datetime | None = None
    status: FlightStatus = "unknown"
    terminal: str | None = None
    gate: str | None = None


class FlightsResponse(BaseModel):
    airport: str
    type: FlightType
    fetched_at: datetime
    cached: bool
    flights: list[Flight]
